### ISSUE TITLE
Trouble COPYing from synlinked directory that was mounted into podman with machine init -v

### ISSUE THREAD

**Issue by wnm3** [containers/podman#13615]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I have a symlinked directory /store/waamount that points to /Users/wnm3/csnext/aiden/ai-c-aiden/common/WAAUXServer/data
I have initialized my machine passing -v /store/waamount:/store/waamount
This has allowed me to run containers from images I'd already built, using the -v /store/waamount:<some_internal_location> without any problems.

However, when attempting to build a new image the Dockerfile gets to a point where it tries to copy content from the /store/waamount into the image and it fails to find the data to be copied. The following is the line that fails:
COPY --chown=1002:0 /store/waamount/TenantInfo.json ./data/TenantInfo.json

Returns this during image build:
STEP 18/22: COPY --chown=1002:0 /store/waamount/TenantInfo.json ./data/TenantInfo.json
Error: error building at STEP "COPY --chown=1002:0 /store/waamount/TenantInfo.json ./data/TenantInfo.json": checking on sources under "/var/tmp/libpod_builder244888758/build": copier: stat: "/store/waamount/TenantInfo.json": no such file or directory

If I perform this command from the directory where I'm performing the build it shows the file exists:
ls -l /store/waamount/TenantInfo.json
-rwxrwxrwx  1 wnm3  staff  13488 Mar 17 12:13 /store/waamount/TenantInfo.json

**Steps to reproduce the issue:**

1. Create a symbolic link from a directory to data that you'd like to copy into the image

-    (on MacOS you need to have created the /store using /etc/synthetic.conf with a line like store Users/wnm3/store) based on [this article](https://derflounder.wordpress.com/2020/01/18/creating-root-level-directories-and-symbolic-links-on-macos-catalina/)
-    cd ~ (/Users/wnm3 on my machine)
-    mkdir data
-    cd data
-    touch TenantInfo.json
-    cd /store
-    ln -s /Users/wnm3/data waamount

2. Initialize podman to mount the /store/waamount directory:

-     podman machine init -v /store/waamount:/store/waamount
-     podman machine start
-     podman machine set --rootful

3. Build an image with a Dockerfile that tries to copy /store/waamount/TenantInfo.json into the image somewhere with a command like (assuming the WORKDIR has a subdirectory data in it:

-     COPY --chown=1002:0 /store/waamount/TenantInfo.json ./data/TenantInfo.json

**Describe the results you received:**
Error: error building at STEP "COPY --chown=1002:0 /store/waamount/TenantInfo.json ./data/TenantInfo.json": checking on sources under "/var/tmp/libpod_builder244888758/build": copier: stat: "/store/waamount/TenantInfo.json": no such file or directory

Though the file being copied clearly exists.

**Describe the results you expected:**
No error and for the build to continue to the next step

**Additional information you deem important (e.g. issue happens only occasionally):**
I am able to reference the mounted /store/waamount when using the -v command in the docker run command to mount it into the container, and it appears to work correctly.

Perhaps this is related to the issue of /store being a mounted directory as well as waamount being a symlink?

**Output of `podman version`:**

```
podman version
Client:       Podman Engine
Version:      4.0.2
API Version:  4.0.2
Go Version:   go1.17.8

Built:      Wed Mar  2 09:04:36 2022
OS/Arch:    darwin/amd64

Server:       Podman Engine
Version:      4.0.2
API Version:  4.0.2
Go Version:   go1.16.14

Built:      Thu Mar  3 09:56:56 2022
OS/Arch:    linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.24.1
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc35.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 4
  distribution:
    distribution: fedora
    variant: coreos
    version: "35"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.15.18-200.fc35.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 4567764992
  memTotal: 6160826368
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.2-1.fc35.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.2
      commit: f6fbc8f840df1a414f31a60953ae514fa497c748
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc35.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 16h 43m 54.89s (Approximately 0.67 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 61
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.0.2
  Built: 1646319416
  BuiltTime: Thu Mar  3 09:56:56 2022
  GitCommit: ""
  GoVersion: go1.16.14
  OsArch: linux/amd64
  Version: 4.0.2
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
brew list podman
/usr/local/Cellar/podman/4.0.2/bin/podman
/usr/local/Cellar/podman/4.0.2/bin/podman-mac-helper
/usr/local/Cellar/podman/4.0.2/bin/podman-remote
/usr/local/Cellar/podman/4.0.2/etc/bash_completion.d/podman
/usr/local/Cellar/podman/4.0.2/libexec/gvproxy
/usr/local/Cellar/podman/4.0.2/share/fish/vendor_completions.d/podman.fish
/usr/local/Cellar/podman/4.0.2/share/man/ (161 files)
/usr/local/Cellar/podman/4.0.2/share/zsh/site-functions/_podman
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by afbjorklund** [comment]
If I recall correctly, the files used with the `COPY` command needs to be in the build context.

**Comment by wnm3** [comment]
I gave up. I could not get podman to work with symbolically linked directories so paid for Docker Enterprise.

**Comment by rhatdan** [comment]
I am not sure what is supposed to happen here. Nor what Docker does.

**Comment by wnm3** [comment]
The use case is I have multiple Eclipse projects that all need test data so I have one real directory of test data and create symbolic links from the other projects to that data directory. Later when I create Docker images from the projects I copy some of the test data into the image and this wasn't working with Podman. I tried loading the entire drive into the podman VM as well as everything under my home directory. Neither worked. Docker has no issue and doesn't require the additional step to explicitly mount something in its VM.

**Comment by rhatdan** [comment]
Is everything located under $HOME?
Did you try `podman machine init -v $HOME:$HOME `?

**Comment by wnm3** [comment]
Yes, the problem was the symbolic links during copy in the Dockerfile that were not working.
